crates/network/src/lib.rs

```diff
@@ -33,24 +34,12 @@ pub use any::AnyNetwork;
 
 pub use alloy_eips::eip2718;
 
-/// A list of transactions, either hydrated or hashes.
-#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
-#[serde(untagged)]
-pub enum TransactionList<T> {
-    /// Hashes only.
-    Hashes(Vec<B256>),
-    /// Hydrated tx objects.
-    Hydrated(Vec<T>),
-    /// Special case for uncle response
-    Uncled,
-}
-
 /// A block response
 #[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
 pub struct BlockResponse<N: Network> {
     #[serde(flatten)]
     header: N::HeaderResponse,
-    transactions: TransactionList<N::TransactionResponse>,
```

Comment:
we should probably do the following while we're here:
- delete this block response as unused and close #441 
- make `TransactionList` generic over `T` if it's not already

Reply:
`TransactionList` is already generic over T, I was removing it in favour of `BlockTransactions` (which does have a generic) from `rpc-types`. 

Also if we're removing `BlockResponse` we can also remove `TransactionList` as it is the only place it is used

Reply:
I'll leave `TransactionsList` as is but would like to change the `Hydrated` enum to `Full` to be consistent with `BlockTransactions`

Reply:
i feel like we're having a miscommunication here so i'm gonna step back

- the 2 types in alloy-network lib.rs should be deleted as they're duplicative
- the types in alloy-rpc-types should have generics added with defaults

Reply:
Sounds great! That's ideal.